pkg/infra/teams/team3/messaging.go

```diff
@@ -1,38 +1,68 @@
 package team3
 
 import (
+	"fmt"
 	"infra/game/agent"
 	"infra/game/commons"
 	"infra/game/message"
+	"sync"
 )
 
 // This is where you must compile your trust message. My example implementation takes ALL agents from the agent map **
 func (a *AgentThree) CompileTrustMessage(agentMap map[commons.ID]agent.Agent) message.Trust {
+	fmt.Println("AGENT 3 COMPOSED: message.Trust")
 
-	// fmt.Println("AGENT 3 COMPOSED: message.Trust")
-
-	keys := make([]commons.ID, len(agentMap))
+	//faireness = the function ids --> reputation number: which is the gossip
+	//send to everyone
+	keys := make([]commons.ID, len(agentMap)+len(a.TSN))
 
 	// ** it extracts the keys (i.e., the IDs) **
 	i := 0
-	for k := range agentMap {
+	for _, k := range a.TSN {
 		keys[i] = k
 		i++
 	}
 
+	for k := range agentMap {
+		keys[i] = k
+		i++
+	}
```

Comment:
I don't understand this. We already have every ID from the agent map...why bother iterating over a second set of agent IDs?